### PR TITLE
Show endpoints on up

### LINKED ISSUE #1908: Show endpoints on `okteto up`
**Is your feature request related to a problem? Please describe.**
I would like to see the endpoints of my application when running `okteto up` to make it easier to access my application.

**Describe alternatives you've considered**
I could use port forwards, but that doesn't cover ingress endpoints

### PR DESCRIPTION
**Pull request by jLopezbarb**
Signed-off-by: Javier López Barba <javier@okteto.com>

Fixes #1908

## Proposed changes
- Show endpoints at the end of up independently if its a vanilla cluster or a okteto cluster
-

### CODE REVIEW

File: cmd/up/up.go [hunk 1]
```diff
@@ -540,5 +575,11 @@ func printDisplayContext(dev *model.Dev, divertURL string) {
 	if divertURL != "" {
 		log.Println(fmt.Sprintf("    %s       %s", log.BlueString("URL:"), divertURL))
 	}
+	if len(endpoints) > 0 {
+		log.Println(fmt.Sprintf("    %s %s", log.BlueString("Endpoints:"), endpoints[0]))
+		for i := 1; i < len(endpoints); i++ {
+			log.Println(fmt.Sprintf("               %s", endpoints[i]))
```
Comment (jmacelroy) [inline cmd/up/up.go:581]: Is there a more straight forward option with `log.Printf`?

File: cmd/up/up.go [hunk 2]
```diff
@@ -509,7 +512,39 @@ func (up *upContext) shutdown() {
 
 }
 
-func printDisplayContext(dev *model.Dev, divertURL string) {
+func (up *upContext) getUpEndpoints(ctx context.Context) []string {
```
Comment (jmacelroy) [inline cmd/up/up.go:515]: All the errors are ignored for the most part. I'm fine with returning an empty slice but should we log the error or are we relying on the original error being logged?

File: cmd/up/up.go [hunk 3]
```diff
@@ -509,7 +512,39 @@ func (up *upContext) shutdown() {
...
+	result := make([]string, 0)
```
Comment (jmacelroy) [inline cmd/up/up.go:516]: Remove this and just return []string{} in the other areas.

File: cmd/up/up.go [hunk 4]
```diff
@@ -509,7 +512,39 @@ func (up *upContext) shutdown() {
...
+	c, _, err := okteto.GetK8sClient()
+	if err != nil {
+		return result
```
Comment (jmacelroy) [inline cmd/up/up.go:519]:
```suggestion
		return []string{}
```

File: cmd/up/up.go [hunk 5]
```diff
@@ -509,7 +512,39 @@ func (up *upContext) shutdown() {
...
+	}
+	endpointList, err := endpoints.GetEndpointsByPodName(ctx, up.Pod.Name, up.Dev.Namespace, c)
+	if err != nil {
+		return result
+	}
+	svcsName := make([]string, 0)
```
Comment (jmacelroy) [inline cmd/up/up.go:525]:
```suggestion
	svcsName := []string{}
```

File: cmd/up/up.go (hunk 1 above)
Comment (jmacelroy) [reply cmd/up/up.go:581]: If so, possibly fix the other logs unless you'd rather just focus on this change.

File: pkg/k8s/endpoints/crud.go [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+}
+
+// GetEndpointsByPodName returns the list of endpoints containing a pod
+func GetEndpointsByPodName(ctx context.Context, podName, namespace string, c *kubernetes.Clientset) ([]apiv1.Endpoints, error) {
+	endpointsList, err := List(ctx, namespace, "", c)
+	if err != nil {
+		return nil, err
+	}
+
+	result := make([]apiv1.Endpoints, 0)
```
Comment (jmacelroy) [inline pkg/k8s/endpoints/crud.go:45]:
```suggestion
	result := []apiv1.Endpoints{}
```
In general, unless you have a known number of allocations it's easier to read and relies on any language optimizations if you use the new form.

**Comment by pchico83** [comment]
@jLopezbarb could you upgrade the description of the PR with a screenshot?

**Comment by pchico83** [comment]
@jLopezbarb let's put this on-hold, or close it for now. I think endpoints should be shown for `deploy`, not `up`, and deploy will be part of up. Otherwise, we are missing endpoints related to my dev container. For exapmple, I would like to see my frontend endpoint when I am developing on the api

**Comment by pchico83** [comment]
Closing it for now until we have the `up/deploy` experience together

**Pull request closed without merging**